src/Sarif.Viewer.VisualStudio.Core/CodeAnalysisResultManager.cs

```diff
@@ -683,6 +685,69 @@ internal SarifLog GetPartitionedLog(IEnumerable<SarifErrorListItem> listItems)
             return partitions[guid];
         }
 
+        internal void AddSuppressionToSarifLog(SuppressionModel suppressionModel)
+        {
+            if (suppressionModel == null || suppressionModel.SelectedErrorListItems?.Any() != true)
+            {
+                return;
+            }
+
+            int runIndex = -1;
+            bool suppressionAdded = false;
+
+            foreach (SarifErrorListItem item in suppressionModel.SelectedErrorListItems)
+            {
+                runIndex = runIndex == -1 ? item.RunIndex : runIndex;
+                if (item.SarifResult != null)
+                {
+                    if (item.SarifResult.Suppressions == null)
+                    {
+                        item.SarifResult.Suppressions = new List<Suppression>();
+                    }
+
+                    var suppression = new Suppression
+                    {
+                        Status = suppressionModel.Status,
+                        Kind = SuppressionKind.External,
+                    };
+
+                    item.SarifResult.Suppressions.Add(suppression);
+                    suppressionAdded = true;
+                }
+            }
+
+            if (runIndex == -1 ||
+                !this.RunIndexToRunDataCache.TryGetValue(runIndex, out RunDataCache dataCache) ||
+                dataCache.SarifLog == null)
+            {
+                return;
+            }
+
+            if (suppressionAdded)
+            {
+                // add empty suppression for results don't have suppression
+                // this is to satisfy sarif spec: either all results have non-null suppressions or have no suppressions
```

Comment:
i personally do not like this.
reason: we are adding  a lot of suppression tags for things that might not have.

imagine a file with 100 results and only one should be suppressed. then, we added 99 empty suppressions and 1 real suppression.

Reply:
Also, add the link to the spec in the comments

Reply:
According to [spec](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317661), either all results have suppression or none of the results have suppression.

>The suppressions values for all result objects in theRun SHALL be either all null or all non-null.
NOTE: The rationale is that an engineering system will generally evaluate all results for suppression, or none of them. Requiring that the suppressions values be either all null or all non-null enables a consumer to determine whether suppression information is available for the run by examining a single result object.

Also the sdk Run.HasSuppressedResults() is based on same logic. vs extension use this API to check if hide suppressed results in error list by default

https://github.com/microsoft/sarif-sdk/blob/45e21b5671100f48807fb74e085342f196ce7513/src/Sarif/Core/Run.cs#L338-L341

Please let me know what you think

Reply:
let's follow the spec and in all results.